src/platform/backends/qemu/linux/qemu_platform_detail_linux.cpp

```diff
@@ -192,27 +195,47 @@ QStringList mp::QemuPlatformDetail::vm_platform_args(const VirtualMachineDescrip
          << QString::fromStdString(fmt::format("tap,ifname={},script=no,downscript=no,model=virtio-net-pci,mac={}",
                                                tap_device_name, vm_desc.default_mac_address));
 
+    auto bridge_helper_path = QCoreApplication::applicationDirPath() + "/bridge_helper";
+
+    for (const auto& extra_interface : vm_desc.extra_interfaces)
+    {
+        opts << "-nic"
+             << QString::fromStdString(fmt::format("bridge,br={},model=virtio-net-pci,mac={},helper={}",
+                                                   extra_interface.id,
+                                                   extra_interface.mac_address,
+                                                   bridge_helper_path));
+    }
+
     return opts;
 }
 
-void mp::QemuPlatformDetail::add_network_interface(VirtualMachineDescription& desc,
-                                                   const NetworkInterface& extra_interface)
+mp::QemuPlatform::UPtr mp::QemuPlatformFactory::make_qemu_platform(const Path& data_dir) const
+{
+    return std::make_unique<mp::QemuPlatformDetail>(data_dir);
+}
+
+bool mp::QemuPlatformDetail::is_network_supported(const std::string& network_type) const
 {
-    // TODO: Do not uncomment the following line when implementing bridging in Linux QEMU. Since implementing it would
-    // yield the same exact implementation than in macOS, we need to move the code out of here, and put it only once
-    // in src/platform/backends/qemu/qemu_virtual_machine.[h|cpp].
-    // desc.extra_interfaces.push_back(net);
+    return network_type == "bridge" || network_type == "ethernet";
+}
 
-    // For the time being, just complain:
-    throw NotImplementedOnThisBackendException("networks");
+bool mp::QemuPlatformDetail::needs_network_prep() const
+{
+    return true;
 }
 
-mp::QemuPlatform::UPtr mp::QemuPlatformFactory::make_qemu_platform(const Path& data_dir) const
+void mp::QemuPlatformDetail::set_authorization(std::vector<NetworkInterfaceInfo>& networks)
 {
-    return std::make_unique<mp::QemuPlatformDetail>(data_dir);
+    const auto& br_nomenclature = MP_PLATFORM.bridge_nomenclature();
+
+    for (auto& net : networks)
+        if (net.type == "ethernet" && net.needs_authorization &&
+            mpu::find_bridge_with(networks, net.id, br_nomenclature))
+            net.needs_authorization = false;
```

Comment:
But where is this ever set to true? 

In LXD, it is set in [here](https://github.com/canonical/multipass/blob/1f67f00527092c7a571529a86986fed4ae7d7add/src/platform/backends/lxd/lxd_virtual_machine_factory.cpp#L64-L67). Seeing as this is really a platform-level thing, I am thinking that should instead be set immediately when reading networks from the system in the Platform. WDYT?

Reply:
Like setting it [here](https://github.com/canonical/multipass/blob/1f67f00527092c7a571529a86986fed4ae7d7add/src/platform/platform_linux.cpp#L152), then?

Reply:
That is what I was thinking, but now I am not sure anymore, because not all backends need that. Namely Virtualbox, which can just connect to an interface without a dedicated switch. I guess the easiest for now is to just add the logic to necessary in here — but set it to `true` as necessary, not to `false` (unless I am missing something).